### PR TITLE
Bump hamcrest/hamcrest to 2.0-dev

### PR DESCRIPTION
**Pull request by davedevelopment**
Refs #464

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "require": {
         "php": ">=5.4.0",
         "lib-pcre": ">=7.0",
-        "hamcrest/hamcrest-php": "~1.1"
+        "hamcrest/hamcrest-php": "~2.0-dev"
```
Comment (aik099) [inline composer.json]: The constraint should say `^2.0@dev` instead. This way:

* you will be checking for a regular version
* for Mockery contributors the dev version of Hamcrest will be used
* for everybody else (who include Mockery as a dependency) stable 2.x version of Hamcrest will be used

Comment (davedevelopment) [inline composer.json]: done

**Comment by aik099** [comment]
Tests pass, so I guess it does solve the problem.

Or maybe we have some code left that orders auto-loader of PHPUnit and Hamcrest to play along? If that is the case, then we need to remove that and use new method, that autoloads Hamcrest global functions where needed in test suite of Mockery.

**Comment by davedevelopment** [comment]
I don't think mockery's test suite was affected, as it only comes in to play if people manually attempt to load phpunit's assert functions.

**Comment by warrickhill** [comment]
Any chance this could be reverted. Getting this when running composer update:
```
Problem 1
    - mockery/mockery dev-master requires hamcrest/hamcrest-php ^2.0@dev -> no matching package found.
```

**Comment by davedevelopment** [comment]
@warrickhill what version of composer are you using? Do you have a custom repository set up for hamcrest?

**Comment by warrickhill** [comment]
I have just run
```composer selfupdate
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Updating to version 593b88e41487f481d4368a4d331146914c1c2981.
    Downloading: 100%         
Use composer self-update --rollback to return to version 03299ff075236be27be356498d6c64def973fe41```

I am getting the problem on both versions. composer install also doesn't work.

No i'm not using a custom repo for hamcrest

**Comment by chrismou** [comment]
@warrickhill Do you have "minimum-stability": "dev" set?  It won't pull through if you're set to stable only

**Comment by davedevelopment** [comment]
Ah yeah, that would be the problem. I think you'll need to adjust the `minimum-stability` setting, or manually require `hamcrest/hamcrest-php: "^2.0@dev"` for now

**Comment by warrickhill** [comment]
both those solutions have worked. I think we will have to manually require hamcrest for now. thanks

**Comment by GrahamCampbell** [comment]
min stability dev, then prefer stable true is often a good combination

**Comment by GrahamCampbell** [comment]
https://github.com/CachetHQ/Cachet/blob/v2.0.4/composer.json#L102-L103

**Pull request merged**